### PR TITLE
Add deprecation warnings for enforce-mountable-secrets annotation

### PR DESCRIPTION
**Pull request by ritazh**
### Description
Add warnings for deprecated `kubernetes.io/enforce-mountable-secrets `annotation. Encourage the use of separate namespaces when isolation is desired.

### Issue

Closes: #

/sig auth
/cc @liggitt

### CODE REVIEW

**Comment by liggitt** [comment]
/lgtm
for technical content

**Review by sftim — COMMENTED** [review]
Thanks for the PR, but:

- please follow the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/)
  - use `note` callouts, not `warning`
  - small but important detail: don't deprecate the annotation except in the well-known list. Deprecate _setting it_.
  - It'll have been deprecated "since Kubernetes v1.32", not "in v1.32+"
- we should also remove or reword all places where we recommend setting the annotation

**Comment by sftim** [comment]
Because this PR adds warnings (as in: warning-level callouts), and we try to be really sparing with these, I suggest we don't merge this yet.

**Comment by sftim** [comment]
/lgtm
for docs

**Comment by sftim** [comment]
Oops. Wrong PR.

File: content/en/docs/concepts/security/secrets-good-practices.md [hunk 1]
```diff
@@ -62,12 +62,16 @@ recommendations include:
 *  Implement audit rules that alert on specific events, such as concurrent
    reading of multiple Secrets by a single user
 
-#### Additional ServiceAccount annotations for Secret management
+#### Additional ServiceAccount annotations for Secret management (deprecated)
```
Comment (sftim) [inline content/en/docs/concepts/security/secrets-good-practices.md:65]: We can use a singular here.
```suggestion
#### ServiceAccount annotation for Secret management (deprecated) {#additional-serviceaccount-annotations-for-secret-management}
```

**Review by sftim — COMMENTED** [review]
/lgtm

One nit; good to merge as is.

File: content/en/docs/concepts/security/secrets-good-practices.md [hunk 2]
```diff
@@ -62,12 +62,16 @@ recommendations include:
 *  Implement audit rules that alert on specific events, such as concurrent
    reading of multiple Secrets by a single user
 
-#### Additional ServiceAccount annotations for Secret management
```
Comment (sftim) [inline content/en/docs/concepts/security/secrets-good-practices.md:65]: Actually, in a page about _good practice_, maybe we should simply be recommending using namespaces as trust boundaries?

File: content/en/docs/concepts/security/secrets-good-practices.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/security/secrets-good-practices.md:65]: The point of having a customized anchor ID is to make it easy to reference this section or subsection. In that spirit, I'd suggest we use a shorter ID which is
unique and clear enough in this page. For example, `#additional-annotation`.

File: content/en/docs/concepts/security/secrets-good-practices.md (hunk 2 above)
Comment (ritazh) [reply content/en/docs/concepts/security/secrets-good-practices.md:65]: Yes, IMO this section should be replaced with the following but I’m not sure how this impacts users who may have bookmarked the old section.

#### Restrict Access for Secrets

Use separate namespaces to isolate access to mounted secrets.

Comment (ritazh) [reply content/en/docs/concepts/security/secrets-good-practices.md:65]: PTAL

**Review by sftim — COMMENTED** [review]
/lgtm

**Comment by ritazh** [comment]
Anything else needed to get this merged?

**Pull request merged**